### PR TITLE
Routing fixes

### PR DESCRIPTION
**Pull request by axdanbol**
Issue - #48 

Description of changes - 

@mentions of the person or team responsible for reviewing proposed changes.

TO-DO checklist before submitting your pull-request - 

- [x] Are all acceptance criteria met ? 
- [x] Has the code been documented properly ?
- [x] Does component css-class structure conform to the Blue Dino style template i.e. top-level element .wrapper, .container and other inside css classes ?
- [x] Have all checks passed ?
- [x] Have you tested inter-browser functionality (should work on the latest Chrome and Firefox browsers) ?

### CODE REVIEW

File: src/app/components/home/home.component.ts [hunk 1]
```diff
@@ -43,25 +67,20 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
...
+    this.initSubject.complete();
   }
 
   ngOnDestroy() {
-    if (this.dataSubscription) {
-      this.dataSubscription.unsubscribe();
-    }
+    this.updateSubject.complete();
+    this.dataSubscription.unsubscribe();
+    this.routerSubscription.unsubscribe();
```
Comment (GagsdName) [inline src/app/components/home/home.component.ts]: check if it is defined, then unsubscribe ?

File: src/app/components/macroscope/macroscope.component.ts [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
 
-  backClick(): void { this.location.back(); }
+  constructor(route: ActivatedRoute) {
+    this.iterationIdSubscription = route.paramMap.subscribe(pmap => {
+      this.iterationId = +pmap.get('iid');
+    });
+  }
+
+  ngOnDestroy() {
+    this.iterationIdSubscription.unsubscribe();
```
Comment (GagsdName) [inline src/app/components/macroscope/macroscope.component.ts]: check if it is defined, then unsubscribe ?

File: src/app/components/home/home.component.ts [hunk 3]
```diff
@@ -26,15 +26,39 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
...
     public readonly modalService: DescriptionModalService,
     private readonly changeDetector: ChangeDetectorRef,
     private readonly dataService: MacroscopeDataService,
-    private readonly route: ActivatedRoute,
-    private readonly router: Router
-  ) { }
+    private readonly router: Router,
+    route: ActivatedRoute
+  ) {
+    const { initSubject, updateSubject } = this;
```
Comment (GagsdName) [inline src/app/components/home/home.component.ts]: spacing looks off here

**Comment by GagsdName** [comment]
Also take a pull from develop

File: src/app/components/home/home.component.ts (hunk 1 above)
Comment (axdanbol) [reply src/app/components/home/home.component.ts]: No need it is always created in ngOnInit

File: src/app/components/macroscope/macroscope.component.ts (hunk 2 above)
Comment (axdanbol) [reply src/app/components/macroscope/macroscope.component.ts]: Always created

File: src/app/components/home/home.component.ts (hunk 3 above)
Comment (axdanbol) [reply src/app/components/home/home.component.ts]: It is not

File: src/app/components/home/home.component.ts (hunk 1 above)
Comment (GagsdName) [reply src/app/components/home/home.component.ts]: Helps in unit testing plus its a good practice. Always better to have checks.

File: src/app/components/macroscope/macroscope.component.ts (hunk 2 above)
Comment (GagsdName) [reply src/app/components/macroscope/macroscope.component.ts]: Helps in unit testing plus its a good practice. Always better to have checks.

File: src/app/components/home/home.component.ts (hunk 3 above)
Comment (GagsdName) [reply src/app/components/home/home.component.ts]: line 41 needs to be indented more

Comment (GagsdName) [reply src/app/components/home/home.component.ts]: and the following lines too I think

Comment (axdanbol) [reply src/app/components/home/home.component.ts]: No, the closing parenthesis should have the same indentation as 'constructor'. The body is the indented once.

File: src/app/components/macroscope/macroscope.component.ts (hunk 2 above)
Comment (axdanbol) [reply src/app/components/macroscope/macroscope.component.ts]: No, unnecessary checks clutter the code. And why would it help in unit testing?

Comment (axdanbol) [reply src/app/components/macroscope/macroscope.component.ts]: Also if that was the case all other member variables should be checked at every usage.

**Pull request merged**